backend/benchmark.py

```
@@ -0,0 +1,431 @@
+
+        print(f"\n[BENCHMARK] Processing batch {batch_id} ({len(batch)} rows)...")
+
+        futures = {RAG_EXECUTOR.submit(process_row, row): row for row in batch}
+
+        for future in as_completed(futures):
+            try:
+                eval_id, results, answer, start = future.result()
+                append_result_row(RESULTS_CSV_PATH, eval_id, results, answer)
+                append_query_timing(eval_id, batch_id, start)
+
+            except Exception as e:
```

Ruff (0.15.15)

[warning] Do not catch blind exception: `Exception`

(BLE001)